database/migrations/2019_02_27_174043_student_course.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class StudentCourse extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('student_course', function (Blueprint $table) {
```

Comment:
maybe esto también debería estar definido en el modelo? como una relación many to many. 

[https://laravel.com/docs/5.8/eloquent-relationships#many-to-many](https://laravel.com/docs/5.8/eloquent-relationships#many-to-many)

Reply:
> As mentioned previously, to determine the table name of the relationship's joining table, Eloquent will join the two related model names in alphabetical order. However, you are free to override this convention.

Por default Laravel crearía una tabla course_user para ambos casos, yo hice un override y lo separé en 'student_course' y 'professor_course' para hacerlo más entendible.